src/Data/MonadicStreamFunction/InternalCore.hs

```diff
@@ -133,3 +132,66 @@ reactimate :: Monad m => MSF m () () -> m ()
 reactimate sf = do
   (_, sf') <- unMSF sf ()
   reactimate sf'
+
+-- Implementation of some common operations for type-class instances, without the instances
+map_msf :: Functor m => (b -> c) -> MSF m a b -> MSF m a c
+map_msf f sf = MSF $ \ a -> fS <$> unMSF sf a where
```

Comment:
An implementation without the `MSF` constructor would be preferable. If you want one with the `MSF` constructor, have a look at https://github.com/ivanperez-keera/dunai/blob/c74aebe596791651edba543998ec85328a58b2e7/src/Data/MonadicStreamFunction/Core.hs#L89, it's already there in a comment.

Reply:
Yeah, that was the one I took as a base.